### PR TITLE
Refactor - Move interfaces of address-lookup-table into the program SDK

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
Unlike the other built-in programs the address-lookup-table does not have a module in the program SDK. Instead other crates which depend on it use its program crate directly, which leads to dependency bloat: E.g. they implicitly also gain the entire program-runtime as a transitive dependency.

#### Summary of Changes
Moves `error`, `state` and `instruction` of address-lookup-table from its program crate into the program SDK.

### CODE REVIEW

File: sdk/program/src/address_lookup_table_account.rs [hunk 1]
```diff
@@ -4,10 +4,8 @@
...
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct AddressLookupTableAccount {
-    pub key: Pubkey,
-    pub addresses: Vec<Pubkey>,
-}
+#[deprecated(
+    since = "1.17.0",
+    note = "Please use `solana_sdk::address_lookup_table::AddressLookupTableAccount` instead"
+)]
+pub use crate::address_lookup_table::AddressLookupTableAccount;
```
Comment (t-nelson) [inline sdk/program/src/address_lookup_table_account.rs:11]: kill the module file in favor of `pub mod address_lookup_table_account { }` in lib.rs, then deprecate the whole module?

File: programs/address-lookup-table/src/lib.rs [hunk 2]
```diff
@@ -2,12 +2,4 @@
 #![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(specialization))]
 #![cfg_attr(RUSTC_NEEDS_PROC_MACRO_HYGIENE, feature(proc_macro_hygiene))]
 
-use solana_program::declare_id;
-
-pub mod error;
```
Comment (t-nelson) [inline programs/address-lookup-table/src/lib.rs:7]: can we add deprecated re-exports for the removals in this file, for the sake of backward compatibility?

**Review by t-nelson — COMMENTED** [review]
much better organization, imo!  just a couple deprecation nits

**Review by t-nelson — COMMENTED** [review]
thanks! i'm r+ now. ping me if no one dissents or approves in 24hrs and i'll approve

**Comment by CriesofCarrots** [comment]
Thanks for preserving the existing public interface

**Review by t-nelson — APPROVED** [review]
:shipit:

**Pull request merged**